src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentTest.kt

```diff
@@ -3537,6 +3558,7 @@ class AssessmentTest : IntegrationTestBase() {
         allocated = assessment.allocatedToUser != null,
         status = status,
         dueAt = assessment.dueAt?.toTimestamp(),
+        probationDeliveryUnitName = (assessment.application as? TemporaryAccommodationApplicationEntity)?.probationDeliveryUnit?.name,
```

Comment:
This this is ok, lots of tests will fail without as?.  If `assessment.application` is not `TemporaryAccommodationApplicationEntity` this returns null due to cast failing.  If `assessment.application` is not null but `probationDeliveryUnit` is null, then null is also returned, which makes sense for applications for which the PDU hasn't been specified (and would therefore need to be null).